sunpy/map/mapbase.py

```diff
@@ -752,6 +749,62 @@ def date(self):
             # UTC is the FITS standard default
             timesys = self.meta.get('timesys', 'UTC')
 
+        return parse_time(time, scale=timesys.lower())
+
+    @property
+    def date_beg(self):
+        """
+        Time of the beginning of the image acquisition.
+
+        Taken from the DATE-BEG FITS keyword.
+        """
+        return self._get_date('date-beg')
+
+    @property
+    def date_end(self):
+        """
+        Time of the end of the image acquisition.
+
+        Taken from the DATE-END FITS keyword.
+        """
+        return self._get_date('date-end')
+
+    @property
+    def date_avg(self):
+        """
+        Average time of the image acquisition.
+
+        Taken from the DATE-AVG FITS keyword if present, otherwise halfway
+        between `date_beg` and `date_end` if both peices of metadata are
+        present.
+        """
+        avg = self._get_date('date-avg')
+        if avg is None:
+            start, end = self.date_beg, self.date_end
+            if start is not None and end is not None:
+                avg = start + (end - start) / 2
+
+        return avg
+
+    @property
+    def date(self):
+        """
+        Image observation time.
+
+        This is the time component of the observer coordiante (if applicable).
+        It is taken from, in order of preference:
+
+        1. `~sunpy.map.GenericMap.date_avg`
+        2. The DATE-OBS FITS keyword
+        3. `~sunpy.map.GenericMap.date_beg`
+        4. `~sunpy.map.GenericMap.date_end`
+        5. The current time
+        """
+        time = self.date_avg
+        time = time or self._get_date('date-obs')
+        time = time or self.date_beg
+        time = time or self.date_end
+
         if time is None:
             if self._default_time is None:
                 warn_metadata("Missing metadata for observation time, "
```

Comment:
This warning message now feels incomplete as there are other ways to suppress this warning than by setting 'DATE-OBS'.

Reply:
I've swapped to DATE-AVG as the recommended way to insert a date, does that seem better?